File: 386.lexicographical-numbers/solution.kt
class Solution {
    fun lexicalOrder(n: Int): List<Int> {
        val res = mutableListOf<Int>()

        fun dfs(k: Int) {
            if (k <= n) {
                res.add(k)
                repeat(10) { i -> dfs(10 * k + i) }
            }
        }

        for (i in 1..9) {
            dfs(i)
        }

        return res
    }
}

File: 386.lexicographical-numbers/solution.go
func lexicalOrder(n int) []int {
    var res []int
    var dfs func(k int) 

    dfs = func(k int) {
        if k <= n {
            res = append(res, k)
            for i := range 10 {
                dfs(10 * k + i)
            }
        }
    }

    for i := 1; i < 10; i++ {
        dfs(i)
    }
    return res
}